### PR TITLE
feat: start new attempt

### PR DESCRIPTION
**Pull request by hlibmykhailov**

### CODE REVIEW

File: src/pages/results/ResultsPage.js [hunk 1]
```diff
@@ -238,6 +265,7 @@ export class ResultsPage extends PureComponent {
               this.renderDownloadCertificateButton()}
             {!isPassed && this.renderContinueCourseButton()}
             {isPassed && !isScormMode && this.renderCloseCourseButton()}
+            {isPassed && !isScormMode && this.renderStartNewAttemptButton()}
```
Comment (oleksandrbulakh) [inline src/pages/results/ResultsPage.js]: !isScormMode - remove

File: src/components/popups/startNewAttempt/StartNewAttempt.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { BUTTON, LINK } from 'constants/components';
+import DescriptionWithBackground from '../descriptionWithBackground';
+import {
+  DescriptionWrapper,
+  TitleElement,
+  ContinueLaterWrapper,
+  ButtonElement
+} from './StartNewAttempt.styled';
+
+export class StartNewAttempt extends Component {
```
Comment (oleksandrbulakh) [inline src/components/popups/startNewAttempt/StartNewAttempt.js]: change to function

File: src/pages/results/ResultsPage.js [hunk 3]
```diff
@@ -62,6 +67,18 @@ export class ResultsPage extends PureComponent {
     this.setState({ npsPopupWasOpened: true });
   }
 
+  openStartNewAttemptPopup = () => {
+    this.props.popupActions.openPopup({
```
Comment (oleksandrbulakh) [inline src/pages/results/ResultsPage.js]: use { courseActions, navigateToIndex } = this.props;

File: src/store/course/actions.js [hunk 4]
```diff
@@ -176,6 +177,48 @@ export const cleanup = () => async (dispatch, getState) => {
   }
 };
 
+export const startNewAttempt = () => async (dispatch, getState) => {
+  dispatch({
```
Comment (oleksandrbulakh) [inline src/store/course/actions.js]: await dispatch(cleanup());
await progressStorage.clearProgress();

await dispatch(launch());

**Comment by YuliaSavchuk** [comment]
What is DescriptionWithBackground popup stands for?

**Comment by YuliaSavchuk** [comment]
The task can be moved to test

**Pull request merged**